### DISCUSSION TITLE
Are Url rewites like in NextJs supported?

### DISCUSSION THREAD

**Discussion by apalumbo** [remix-run/remix#2091] [Q&A] [answered]
We are considering to move our application to remix from NextJs. We are using url rewrite for a couple of features like:

- Proxy traffic to another server (nginx for custom redirects , nextjs has limits on the number of redirects supported)
- rewriting some urls (regexr based) to the filesystem routing like /:locale/:slug(.*-cp)

There is a way to support those use cases in remix?

**Comment by sergiodxa** [comment, marked as answer]
The first one can be setup with a resource route or a proxy in your server (outside Remix), for example in Express or using Vercel rewrites.

The second one can be setup using the remix.config.js routes function to define custom routes outside the app/routes file system based router.
